src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php

```diff
@@ -373,17 +371,33 @@ private function isMaxDepthReached(array $attributesMetadata, $class, $attribute
     }
 
     /**
-     * Gets the cache key to use.
+     * Overwrite to update the cache key for the child.
+     *
+     * {@inheritdoc}
+     */
+    protected function createChildContext(array $parentContext, $attribute)
+    {
+        $context = parent::createChildContext($parentContext, $attribute);
+        // format is already included in the cache_key of the parent.
+        $context['cache_key'] = $this->getAttributeCacheKey('', $context);
```

Comment:
this means that instead of calculating one cache key, we now calculate one for each nested object. this might lead to a performance regression, but the cache key ignores context fields that are irrelevant for the attributes cache.

Reply:
@dunglas if we had the $format available here, the key would not change if there are no changes in the context object - as is, if we don't have any context changes between parent and children, we have 2 keys, one for the root object and one for all children.

either way if a manual key was specified, it is always overwritten here.

Reply:
You can pass it, retrieve it using func_get_args() and deprecate not passing this parameter.

Reply:
good idea, changed